README.md

```diff
@@ -22,23 +22,121 @@ At this point you can either run the repo using Docker or directly with Ruby.
 
+## Publishing to JSBin
+
+This configuration is not required for basic documentation creation and modification, so most
+editors can simply skip to [Forking and running locally](#forking-and-running-locally).
+
+If you will be creating or modifying code within this repository that is to be uploaded automatically
+to [JSBin](https://jsbin.ably.io/), for our "Try it now code editor", then you will need to create yourself a JSBin config file:
+
+### Ably staff members
+
+To obtain the API key you need to run this command in the terminal, you must ensure you have installed the hero-cli tools (Instructions on [installing the Heroku toolbelt](https://devcenter.heroku.com/articles/heroku-cli) )
+
+```bash
+$ heroku pg:psql -a ably-jsbin -c "select name, api_key from ownership"
+```
+
+When you execute this command a browser window will open and you will need to login using your **Ably email address**, only Ably staff members will be granted permission to access this resource.
+
+If you are a staff member and you are receiving this message
+
+```bash
+$ heroku pg:psql -a ably-jsbin -c "select name, api_key from ownership"
+ ▸    You do not have permission to view addon resources on ably-jsbin. You need to
+ ▸    have the deploy or operate permission on this app.
+```
+
+then ask your admin to check you have permission on Heroku, then re-authenticate with.
+
+```bash
+$ heroku login
+```
+
+If everything goes smoothly you will be given the API key.
+
+```
+ name |      api_key
+------+-------------------
+ ably | 12345678901234567
+(1 row)
+```
+
 ### Running with Ruby
 
 Using Ruby, run the following command:
 
-* `bundle install` 
+```bash
+$ cp config/jsbin_config.example.yaml config/jsbin_config.yaml
+```
+
+### Compile the JSBin code snippets
+
+Once you have finished creating your code snippet you will need to add the new assets to the existing ones. This is essentially a table of hashed documents and corresponding file names, [./jsbins.yaml](./jsbins.yaml) if you are interested in the entire list.
+
+```bash
+$ bundle exec nanoc compile
+```
+
+This will output a huge list that looks similar to this, as each file is checked
+
+```bash
+Published new JsBin for hub-product/http-javascript at https://jsbin.ably.io:443/ujehow/1/edit?javascript,live
+Copied https://jsbin.ably.io:443/ujehow/1/edit?javascript,live to clipboard
+	create  [0.49s]  output/code/realtime/channel-deltas-sse/index.html
+    update  [1.10s]  output/sse/index.html
+    update  [1.02s]  output/concepts/socketio/index.html
+```
+
+This process has added a record to `jsbin.yaml` Please notice the `ID` string in the Published URL (above) which is`ujehow` By hashing the contents, we don't need to contact the JSBin API on every static build. We only do this if a hash does not exist, and thus we need to notify JSBin to create a JSBin which contains the HTML, CSS & JS.
+
+```yaml
+---
+jsbin_hash:
+	Rh81oraUHHJ3PrrvwdJAOrVBqA8=: ujehow <-- hash: ID
```

Comment:
The red blocks here and below are a little odd, can we remove them?

Reply:
yeah, it is and it doesn't show on the rendered MD. sure I'll refactor that.